node-data/src/encoding.rs

```diff
@@ -234,6 +234,53 @@ impl Serializable for StepVotes {
     }
 }
 
+impl Serializable for IterationsInfo {
+    fn write<W: Write>(&self, w: &mut W) -> io::Result<()> {
+        let count = self.cert_list.len() as u8;
+        w.write_all(&count.to_le_bytes())?;
+
+        let count_of_some =
+            self.cert_list.iter().filter(|cert| cert.is_some()).count() as u8;
+
+        w.write_all(&count_of_some.to_le_bytes())?;
+
+        for i in 0..self.cert_list.len() as u8 {
+            if let Some(cert) = &self.cert_list[i as usize] {
+                w.write_all(&i.to_le_bytes())?;
+                cert.write(w)?;
+            }
+        }
+
+        Ok(())
```

Comment:
> ps: will we ever have an iteration set to None? (I don't think, since we now include failed iterations without quorum too)

theoretically yes, if a node is offline/eclipsed it might miss all votes for one or more iterations, then time-out and move on.

> I don't think this would be a right approach.
> Serializing total count and count_of_some would loose the info about which cert is missing.
> Instead I would serialize all the option, with a byte prefix that indicates if there is a Some or None

It does not miss anything. Each certificate is serialized as (pos, certificate_bytes). There is also unit test. If you insist I'll refactor it.

Reply:
You're right, just seen the `pos` serialization, my bad.

We can optimize it later on, removing the need for both pos and count_some and using the option serialization